libraries/FSTools/FSTools.cpp

```diff
@@ -0,0 +1,247 @@
+
+	 const layout layout_2m64    (0x403F0000, 0x403FB000, 0x100, 0x1000 );
+	 const layout layout_2m128   (0x403E0000, 0x403FB000, 0x100, 0x1000 );
+	 const layout layout_2m256   (0x403C0000, 0x403FB000, 0x100, 0x1000 );
+	 const layout layout_2m512   (0x40380000, 0x403FA000, 0x100, 0x2000 );
+	 const layout layout_2m1m    (0x40300000, 0x403FA000, 0x100, 0x2000 );
+
+	 const layout layout_4m1m    (0x40500000, 0x405FA000, 0x100, 0x2000 );
+	 const layout layout_4m2m    (0x40400000, 0x405FA000, 0x100, 0x2000 );
+	 const layout layout_4m3m    (0x40300000, 0x405FA000, 0x100, 0x2000 );
+
+	 const layout layout_8m6m    (0x40400000, 0x409FA000, 0x100, 0x2000 );
+	 const layout layout_8m7m    (0x40300000, 0x409FA000, 0x100, 0x2000 );
+
+	 const layout layout_16m14m  (0x40400000, 0x411FA000, 0x100, 0x2000 );
+	 const layout layout_16m15m  (0x40300000, 0x411FA000, 0x100, 0x2000 );
+
+
+};
+
+FSTools::FSTools()
+{
+
+}
+
+FSTools::~FSTools()
+{
+	reset(); 
+}
+
+
+bool FSTools::attemptToMountFS(fs::FS & fs)
+{
+  LittleFSConfig littleFSCfg(false);
+  SPIFFSConfig SPIFFSCfg(false); 
+  //  try to apply the "safe" no format config to the FS... doesn't matter which.. just need one to apply.. 
+  if (!fs.setConfig(littleFSCfg) && ! fs.setConfig(SPIFFSCfg) )
+  {
+  	return false; 
+  }
+  return fs.begin(); 
+}
+
+
+bool FSTools::mountAlternativeFS(  FST::FS_t type, const FST::layout & layout, bool keepMounted )
+{
+	FSConfig * pCfg{nullptr};
+    LittleFSConfig littleFSCfg(false);
+    SPIFFSConfig SPIFFSCfg(false); 
+    reset();
+
+	switch (type) {
+		case FST::SPIFFS : {
+			_pFS.reset( new FS(FSImplPtr(new spiffs_impl::SPIFFSImpl    (_getStartAddr(layout) , _getSize(layout) , layout.page, layout.block, 5)))  );
+			pCfg = &SPIFFSCfg; 
+			break;
+		}
+		case FST::LITTLEFS : {
+			_pFS.reset( new FS(FSImplPtr(new littlefs_impl::LittleFSImpl(_getStartAddr(layout) , _getSize(layout) , layout.page, layout.block, 5)))  );
+			pCfg = &littleFSCfg; 
+			break;
+		}
+	};
+
+	if (_pFS && pCfg && _pFS->setConfig(*pCfg) && _pFS->begin()) {
+		if (!keepMounted) {
+			_pFS->end();
+		}
+		_mounted = true; 
+		_layout = &layout;
+		return true; 
+	}
+
+	if (_pFS) {
+		_pFS.reset();
+	}
+	_mounted = false; 
+	return false; 
+
+}; 
+
+
+bool FSTools::mounted()
+{
+	return _mounted; 
+};
+
+
+
+
+bool FSTools::moveFS(fs::FS & destinationFS)
+{
+    uint32_t sourceFileCount = 0; 
+    uint32_t sourceByteTotal = 0; 
+    bool result = false; 
+
+	//  do not init new fs... until old one is copied to test... 
+	if (!_mounted || !_pFS) {
+		//Serial.println("Source FS not mounted"); 
+		return false; 
+	}
+
+
+  	uint32_t startSector = (ESP.getSketchSize() + FLASH_SECTOR_SIZE - 1) & (~(FLASH_SECTOR_SIZE - 1)); 
+  	//uint32_t endSector = (uint32_t)&_FS_start - FST::lowestSPIFFSstartAddr;
+  	uint32_t lowestFSStart = 0x40300000;
+  	
+  	 if (_layout) {
+  	 	lowestFSStart = _layout->startAddr;  	
+  		//Serial.printf("**  _layout->startADDR = 0x%08x\n", _layout->startAddr ); 
```

Comment:
For debug messages, why not use `#ifdef DEBUG_ESP_CORE` bracketing instead of commenting them out?